File: backend/app/schemas/league.py
from pydantic import BaseModel
from typing import Optional, List
from enum import Enum
from datetime import datetime

class GameType(str, Enum):
    survivor = "survivor"
    love_island = "love_island"
    big_brother = "big_brother"
    traitors = "traitors"

class MemberRole(str, Enum):
    owner = "owner"
    admin = "admin"
    member = "member"

class LeagueSettingsBase(BaseModel):
    max_members: int = 20
    is_private: bool = True

class LeagueSettingsOut(LeagueSettingsBase):
    id: int

    class Config:
        orm_mode = True

class LeagueBase(BaseModel):
    name: str
    game_type: GameType

class LeagueCreate(LeagueBase):
    settings: Optional[LeagueSettingsBase] = None

class LeagueOut(LeagueBase):
    id: int
    created_at: datetime  # changed from str to datetime
    owner_id: int
    join_code: str  # add join_code to schema
    settings: Optional[LeagueSettingsOut]

    class Config:
        orm_mode = True
        from_attributes = True  # for Pydantic v2 compatibility

class LeagueMembershipOut(BaseModel):
    id: int
    league_id: int
    user_id: int
    role: MemberRole
    joined_at: datetime  # changed from str to datetime

    class Config:
        orm_mode = True
        from_attributes = True  # for Pydantic v2 compatibility

File: frontend/src/types/league.ts
// src/types/league.ts
export type GameType = "survivor" | "love_island" | "big_brother" | "traitors";
export type MemberRole = "owner" | "admin" | "member";

export interface LeagueSettingsOut {
  id: number;
  max_members: number;
  is_private: boolean;
}

export interface LeagueOut {
  id: number;
  name: string;
  game_type: GameType;
  created_at: string;
  owner_id: number;
  join_code: string; // Added join_code for LeagueDetails
  settings?: LeagueSettingsOut;
}

export interface LeagueMembershipOut {
  id: number;
  league_id: number;
  user_id: number;
  role: MemberRole;
  joined_at: string;
}
